packages/Checkbox/src/Checkbox.styles.js

```diff
@@ -0,0 +1,138 @@
+import { css } from "styled-components";
+import { spacer, toInt, fontSizeValue, lineHeightValue } from "@paprika/stylers/lib/helpers";
+import { boxSizingStyles, visuallyHidden } from "@paprika/stylers/lib/includes";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import tokens from "@paprika/tokens";
+
+const labelSizeMap = {
+  [ShirtSizes.SMALL]: -1,
+  [ShirtSizes.MEDIUM]: 0,
+  [ShirtSizes.LARGE]: 0,
+};
+
+const iconSizeMap = {
+  [ShirtSizes.SMALL]: -3,
+  [ShirtSizes.MEDIUM]: -1,
+  [ShirtSizes.LARGE]: 0,
+};
+
+const checkboxSizeMap = {
+  [ShirtSizes.SMALL]: 2,
+  [ShirtSizes.MEDIUM]: 2.5,
+  [ShirtSizes.LARGE]: 3,
+};
+
+const labelTopOffsetMap = {
+  [ShirtSizes.SMALL]: 0,
+  [ShirtSizes.MEDIUM]: 1,
+  [ShirtSizes.LARGE]: 3,
+};
+
+const getSpacing = size => spacer(checkboxSizeMap[size]);
+const checkBoxSize = ({ size }) => getSpacing(size);
+const checkBoxHalfSize = ({ size }) => `${toInt(getSpacing(size)) / 2}px`;
+
+const labelSizeValue = ({ size }) => `${fontSizeValue(labelSizeMap[size])}px`;
+const iconSizeValue = ({ size }) => `${fontSizeValue(iconSizeMap[size])}px`;
+
+const labelPadding = ({ hasLabel, size }) => {
+  const top = `${labelTopOffsetMap[size]}px`;
+  const left = hasLabel ? `${toInt(getSpacing(size)) + toInt(tokens.space)}px` : getSpacing(size);
+  return `${top} 0 0 ${left}`;
+};
+
```

Comment:
If everyone else is fine with it, then so be it, but I still dont like all this. Can't it just be something like:

```
switch(size) {
  case ShirtSizes.SMALL:
    $fontSize = font-scale();
    $paddingTop = '4';
    $height = '10';
    break;
  case ShirtSizes.MEDIUM:
    $fontSize = font-scale(1);
    $paddingTop = '0';
    $height = '14';
    break;
  case ShirtSizes.LARGE:
    $fontSize = font-scale(2);
    $paddingTop = '0';
    $height = '15';
    break;
}
```
Then use `fontSize, paddingTop, height` in the css below?

Reply:
I think for now we can go with the approach we have, perhaps refactor later if we think is really too complex. I'm not sure how your `switch would work` if we are needing those styles in on different elements either..

Reply:
any reason to postpone the refactor @tristanjasper ? I think we could achieve the same either with a switch or an object and help with clarity. 

```js
const styles = {
     [ShirtSizes.SMALL]: {},
     [ShirtSizes.MEDIUM]: {},
     [ShirtSizes.LARGE]: {},
}
```

I will let @mikrotron and you to decide if worth to refactor it now or later.

Reply:
@nahumzs @jamiek-acl @mikrotron I've refactored. Let me know if any further changes required.

Reply:
I think that's more clear